TodoApp/tests/TodoApp.Api.Tests/Controllers/TodosControllerTests.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using NSubstitute;
+using NUnit.Framework;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Web.Http;
+using System.Web.Http.Hosting;
+using System.Web.Http.Results;
+using System.Web.Http.Routing;
+using TodoApp.Api.Controllers;
+using TodoApp.Contracts.Helpers;
+using TodoApp.Contracts.Models;
+using TodoApp.Contracts.Repositories;
+using Assert = NUnit.Framework.Assert;
+using CollectionAssert = NUnit.Framework.CollectionAssert;
+
+namespace TodoApp.Api.Tests.Controllers
+{
+    [TestFixture]
+    class TodosControllerTests
+    {
+        private TodosController _controller;
+        private ITodoRepository _mockRepo;
+        private IUriHelper _uriHelper;
+        private Todo _mockTodo;
+        private List<Todo> _mockTodos;
+        private readonly Guid _guid = new Guid("38f61793-bf01-48ae-8e00-ccee139adba2");
+
+        [SetUp]
+        public void Init()
+        {
+            _mockRepo = Substitute.For<ITodoRepository>();
+
+            _uriHelper = Substitute.For<IUriHelper>();
+            _uriHelper.BuildUriForPostTodo(Arg.Any<Guid>())
+                .Returns(parameters => new Uri($"/localhost/todos/{parameters.Arg<Guid>()}", UriKind.Relative));
+
+            _controller = new TodosController(_mockRepo, _uriHelper);
+
+            _mockTodo = new Todo()
+            {
+                Id = _guid,
+                Value = "Make more coffee"
+            };
+
+            _mockTodos = new List<Todo>
+            {
+                new Todo() {Id = new Guid("2e2253c5-4bdb-45d8-8cbf-1a24e9b04d1c"), Value = "Make coffee"},
```

Comment:
If you're using the default constructor and object initializer, you can omit the parentheses. Looks super sexy that way.